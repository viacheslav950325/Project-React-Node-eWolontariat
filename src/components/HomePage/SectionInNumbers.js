import { Box, Button,Typography } from "@material-ui/core";
import LogoPomocny from "../../assets/img/logo-pomocny.svg";
import ImgGlobe from "../../assets/img/globe-mask.svg";

const SectionInNumbers = () => {
    return(
        <Box id={"section-in-numbers"}
            display={"flex:"}
            height = {"100%"}
            justifyContent={"center"}
            alignItems={"center"}
        >
            <Box sx= {{ display: 'flex', flexDirection: 'column', alignItems: "center"}} >
                <Box component="img" height={'60px'} alt="Logo pomocny.pl" src={LogoPomocny}/>
                <Typography variant="h2">w liczbach</Typography>
            </Box>
            <Box display={'flex'} padding={'4rem'} justifyContent={'space-evenly'} mr={'2rem'}>
                <Box alignSelf={'center'} textAlign={'center'}>
                    <Typography  className={"bigNumber"}>100</Typography>
                    <Typography variant="caption">Tylu osobom pomogliśmy</Typography>
                </Box>
                <Box alignSelf={'center'} textAlign={'center'}>
                    <Typography  className={"bigNumber"}>12334</Typography>
                    <Typography variant="caption">Tyle zadań zakończyło się sukcesem</Typography>
                </Box>
                <Box alignSelf={'center'} textAlign={'center'}>
                    <Typography  className={"bigNumber"}>345</Typography>
                    <Typography variant="caption">Tyle zgłosiło się wolontariuszy</Typography>
                </Box>
                
            </Box>
            <Box display={'flex'} padding={'4rem'} justifyContent={'space-evenly'} mr={'2rem'}>
                <Box alignSelf={'center'}>
                    <Typography variant={'body1'} paragraph={true}>
                        Imponujące? Dołącz do nas już dziś i bądź pomocny!
                    </Typography>
                    <Button variant="contained" color='success' href={"/"}>Zostań wolontariuszem</Button>

                </Box>
                <Box component="img" maxWidth={'400px'} alt="" src={ImgGlobe}/>
            </Box>
        </Box>
    )
}

export default SectionInNumbers;