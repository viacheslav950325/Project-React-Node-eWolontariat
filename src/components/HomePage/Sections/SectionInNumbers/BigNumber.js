import * as React from "react";
import { Typography } from "@material-ui/core";
import {makeStyles} from '@material-ui/core';
import { useState } from "react";
import CountUp from 'react-countup';
import VisibilitySensor from 'react-visibility-sensor-v2';
const useStyles = makeStyles({
    bigNumber: {
        fontSize: '4rem',
        fontWeight: '700',
        color: '#4DAF7B',
        height: 100 
    }
});

const BigNumber = React.forwardRef(function BigNumber({end = Number}, ref) {
        const classes = useStyles();
        const [loadingNumber, setLoading] = useState(false);
        const onStart = () => {setLoading(true)};
        const onEnd = () => {setLoading(false)};

        const containerProps = {
            'aria-busy': loadingNumber
        }

        return (
            
            <VisibilitySensor partialVisibility offset={{ bottom: 200 }}>
                {({ sensorRef,isVisible }) => (
                    <Typography className={classes.bigNumber} ref={sensorRef}>
                        {isVisible ?  <CountUp end={end} onStart={onStart} onEnd={onEnd} containerProps={containerProps} /> : null}
                    </Typography>
                )}
            </VisibilitySensor>
            
        );
    }
);

export default BigNumber;