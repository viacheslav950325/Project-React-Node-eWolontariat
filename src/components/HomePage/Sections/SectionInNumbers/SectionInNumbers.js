import { Box, Typography } from "@material-ui/core";
import LogoPomocny from "../../../../assets/img/logo-pomocny.svg";
import ImgGlobe from "../../../../assets/img/globe-mask.svg";
import LogoSignet from "../../../../assets/img/hand-peace-solid.svg";
import CustomButton from "../../../CustomButton";
import BigNumber from "./BigNumber";

const SectionInNumbers = () => {
   
    return(
        <Box id={"section-in-numbers"}
            display={"flex:"}
            height = {"100%"}
            justifyContent={"center"}
            alignItems={"center"}
            p={"4rem"}
        >
            <Box sx= {{ display: 'flex', flexDirection: 'column', alignItems: "center"}} >
                <Box component="img" height={'80px'} alt="Logo pomocny.pl" src={LogoPomocny}/>
                <Typography variant="h1" style={{marginTop: 0}}>w liczbach</Typography>
            </Box>
            <Box display={'flex'} padding={'4rem'} justifyContent={'space-evenly'} mr={'2rem'}>
                <Box alignSelf={'center'} textAlign={'center'}>
                    <BigNumber end={100}/>
                    <Typography variant="body1">Tylu osobom pomogliśmy</Typography>
                </Box>
                <Box alignSelf={'center'} textAlign={'center'}>
                    <BigNumber end={1230}/>
                    <Typography variant="body1">Tyle zadań zakończyło się sukcesem</Typography>
                </Box>
                <Box alignSelf={'center'} textAlign={'center'}>
                    <BigNumber end={560}/>
                    <Typography variant="body1">Tyle zgłosiło się wolontariuszy</Typography>
                </Box>    
            </Box>
            <Box display={'flex'} padding={'4rem'} justifyContent={'space-evenly'} mr={'2rem'}>
                <Box display={'flex'} flexDirection={'column'} alignSelf={'center'} justifyContent={'center'} alignItems={'center'}>
                    <Typography variant={'body1'} paragraph={true} style={{fontSize: '2.3rem', margin: '0 2rem 2rem 2rem'}} align={'center'}>
                        Imponujące? Dołącz do nas już dziś i bądź pomocny!
                    </Typography>
                    <CustomButton variant="contained" color="tertiary" size={'medium'} href={"/VolunteerForm"} style={{maxWidth: '50%'}}>Zostań wolontariuszem</CustomButton>
                </Box>
                <Box component="img" maxWidth={'100px'} alt="" src={LogoSignet} alignSelf={'flex-end'}/>
                <Box component="img" maxWidth={'460px'} alt="" src={ImgGlobe}/>
            </Box>
        </Box>
    )
}

export default SectionInNumbers;