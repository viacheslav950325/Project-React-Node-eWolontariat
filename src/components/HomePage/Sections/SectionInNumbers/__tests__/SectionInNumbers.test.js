import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import SectionInNumbers from "../SectionInNumbers";

describe("SectionInNumbers",()=>{
    it("Should render SectionInNumbers correctly",()=>{
        const wrapper = shallow(<SectionInNumbers/>);
        expect(toJson(wrapper)).toMatchSnapshot();

    });
});