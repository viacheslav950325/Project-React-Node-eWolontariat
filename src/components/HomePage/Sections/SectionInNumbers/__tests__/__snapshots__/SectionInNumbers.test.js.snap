// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionInNumbers Should render SectionInNumbers correctly 1`] = `
<Styled(MuiBox)
  alignItems="center"
  display="flex:"
  height="100%"
  id="section-in-numbers"
  justifyContent="center"
>
  <Styled(MuiBox)
    sx={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <Styled(MuiBox)
      alt="Logo pomocny.pl"
      component="img"
      height="80px"
      src="logo-pomocny.svg"
    />
    <WithStyles(ForwardRef(Typography))
      style={
        Object {
          "marginTop": 0,
        }
      }
      variant="h1"
    >
      w liczbach
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
  <Styled(Styled(MuiBox))>
    <Styled(MuiBox)
      alignSelf="center"
      textAlign="center"
    >
      <ForwardRef(BigNumber)
        end={100}
      />
      <WithStyles(ForwardRef(Typography))
        variant="body1"
      >
        Tylu osobom pomogliśmy
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      alignSelf="center"
      textAlign="center"
    >
      <ForwardRef(BigNumber)
        end={1230}
      />
      <WithStyles(ForwardRef(Typography))
        variant="body1"
      >
        Tyle zadań zakończyło się sukcesem
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      alignSelf="center"
      textAlign="center"
    >
      <ForwardRef(BigNumber)
        end={560}
      />
      <WithStyles(ForwardRef(Typography))
        variant="body1"
      >
        Tyle zgłosiło się wolontariuszy
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
  </Styled(Styled(MuiBox))>
  <Styled(Styled(MuiBox))>
    <Styled(MuiBox)
      alignItems="center"
      alignSelf="center"
      display="flex"
      flexDirection="column"
      justifyContent="center"
    >
      <Styled(WithStyles(ForwardRef(Typography)))
        align="center"
        paragraph={true}
        variant="body1"
      >
        Imponujące? Dołącz do nas już dziś i bądź pomocny!
      </Styled(WithStyles(ForwardRef(Typography)))>
      <ForwardRef(CustomButton)
        color="tertiary"
        href="/VolunteerForm"
        size="medium"
        style={
          Object {
            "maxWidth": "50%",
          }
        }
        variant="contained"
      >
        Zostań wolontariuszem
      </ForwardRef(CustomButton)>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      alignSelf="flex-end"
      alt=""
      component="img"
      maxWidth="100px"
      src="hand-peace-solid.svg"
    />
    <Styled(MuiBox)
      alt=""
      component="img"
      maxWidth="460px"
      src="globe-mask.svg"
    />
  </Styled(Styled(MuiBox))>
</Styled(MuiBox)>
`;
